salt/beacons/__init__.py

```diff
@@ -42,23 +43,40 @@ def process(self, config):
         for mod in config:
             if mod == 'enabled':
                 continue
-            if 'enabled' in config[mod] and not config[mod]['enabled']:
-                log.trace('Beacon {0} disabled'.format(mod))
-                continue
-            elif 'enabled' in config[mod] and config[mod]['enabled']:
-                # remove 'enabled' item before processing the beacon
-                del config[mod]['enabled']
+
+            # Convert beacons that are lists to a dict to make processing easier
+            current_beacon_config = None
+            if isinstance(config[mod], list):
+                current_beacon_config = {}
+                list(map(current_beacon_config.update, config[mod]))
+            elif isinstance(config[mod], dict):
+                salt.utils.warn_until(
+                    'Boron',
```

Comment:
This deprecation path needs to be much longer than just Boron, since that's the next release on this branch. I recommend at least 'Carbon' if not 'Nitrogen'.